### PR TITLE
Monix removal

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: backend/src/main/scala/com/softwaremill/bootzooka/email/EmailService.scala [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
     extends EmailScheduler
     with StrictLogging {
 
   def apply(data: EmailData): ConnectionIO[Unit] = {
     logger.debug(s"Scheduling email to be sent to: ${data.recipient}")
-    idGenerator.nextId().to[ConnectionIO].flatMap(id => emailModel.insert(Email(id, data)))
+    idGenerator
+      .nextId[Email]()
+      .map(id => emailModel.insert(Email(id, data)))
+      .unsafeRunSync()
```
[the remark this answers is not in the document]
Comment (Pask423) [reply backend/src/main/scala/com/softwaremill/bootzooka/email/EmailService.scala:27]: During my research of topics I found out that doing it is such way, using `.to` method is probably impossible.
Currently suggested way to use `WeakAsync.liftK[IO, ConnectionIO]` but it unfortunately uses `unsafeToFuture` and we want to preserve context  - `fiber local`, so we cannot use any `unsafe`
All of this is caused mostly by updates in Cats Effect type type-class hierarchy.

File: backend/src/main/scala/com/softwaremill/bootzooka/email/EmailService.scala [hunk 2]
```diff
@@ -40,19 +41,19 @@ class EmailService(emailModel: EmailModel, idGenerator: IdGenerator, emailSender
       emailModel.count().transact(xa).map(_.toDouble).map(Metrics.emailQueueGauge.set)
     }
 
-    Task.parZip2(sendProcess, monitoringProcess)
+    sendProcess.flatMap(r => monitoringProcess.flatMap(d => IO(r, d)))
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/email/EmailService.scala:44]: that will work differenlty. Previously, we were runing send & monitor *in parallel*. Now, they would run sequentially - causing monitoring never to run, as send never finishes. So we need to change this back to parallel

File: backend/src/main/scala/com/softwaremill/bootzooka/http/EndpointsToRoutes.scala [hunk 3]
```diff
@@ -61,11 +69,11 @@ class EndpointsToRoutes(http: Http, apiContextPath: String) {
...
   /** Interprets the given endpoint descriptions as docs, and returns http4s routes which expose the documentation using Swagger.
     */
-  def toDocsRoutes(es: ServerEndpoints): HttpRoutes[Task] = {
+  def toDocsRoutes(es: ServerEndpoints): HttpRoutes[IO] = {
     val openapi = OpenAPIDocsInterpreter()
       .serverEndpointsToOpenAPI(es.toList, "Bootzooka", "1.0")
       .servers(List(Server(s"$apiContextPath", None)))
-    val yaml = openapi.toYaml
-    new SwaggerHttp4s(yaml).routes[Task]
+    val swaggerRoutes: List[ServerEndpoint[_, _, _, Any, IO]] = SwaggerUI[IO](openapi.toYaml)
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/http/EndpointsToRoutes.scala:76]: tapir needs updating to 0.19-final :)

Comment (adamw) [reply backend/src/main/scala/com/softwaremill/bootzooka/http/EndpointsToRoutes.scala:76]: also maybe we can use the bundle: https://tapir.softwaremill.com/en/latest/docs/openapi.html#using-swagger

**Comment by adamw** [comment]
There's quite a lot of compiler warnings - maybe we could fix them. Also, needs a rebase.

File: backend/src/main/scala/com/softwaremill/bootzooka/Main.scala [hunk 4]
```diff
@@ -1,17 +1,15 @@
...
 import com.softwaremill.bootzooka.metrics.Metrics
 import com.typesafe.scalalogging.StrictLogging
 import doobie.util.transactor
-import monix.eval.Task
-import monix.execution.Scheduler.Implicits.global
 import sttp.client3.SttpBackend
 
 object Main extends StrictLogging {
   def main(args: Array[String]): Unit = {
-    CorrelationId.init()
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/Main.scala:14]: we need to migrate correlation id support as well:) probably implementing https://github.com/softwaremill/monix-correlation-id using fiber-locals inside bootzooka for now, we might extract it as a separate repository later

File: backend/src/main/scala/com/softwaremill/bootzooka/email/EmailService.scala [hunk 5]
```diff
@@ -1,54 +1,55 @@
...
+      .flatMap(id => emailModel.insert(Email(id, data)))
   }
 
-  def sendBatch(): Task[Unit] = {
+  def sendBatch(): IO[Unit] = {
     for {
       emails <- emailModel.find(config.batchSize).transact(xa)
       _ = if (emails.nonEmpty) logger.info(s"Sending ${emails.size} emails")
-      _ <- Task.sequence(emails.map(_.data).map(emailSender.apply))
+      _ <- IO(emails.map(_.data).map(emailSender.apply))
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/email/EmailService.scala:27]: what happened to .sequence? This won't work, we're constructing an `IO[IO[Unit]]` and ignoring the inner result

File: backend/src/main/scala/com/softwaremill/bootzooka/email/sender/DummyEmailSender.scala [hunk 6]
```diff
@@ -14,7 +14,7 @@ object DummyEmailSender extends EmailSender with StrictLogging {
     sentEmails.clear()
   }
 
-  override def apply(email: EmailData): Task[Unit] = Task {
+  override def apply(email: EmailData): IO[Unit] = IO.pure {
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/email/sender/DummyEmailSender.scala:17]: This definitely isn't pure! Fixing

File: backend/src/test/scala/com/softwaremill/bootzooka/test/BaseTest.scala [hunk 7]
```diff
@@ -1,10 +1,8 @@
 package com.softwaremill.bootzooka.test
 
-import com.softwaremill.bootzooka.infrastructure.CorrelationId
 import org.scalatest.flatspec.AnyFlatSpec
 import org.scalatest.matchers.should.Matchers
 
 trait BaseTest extends AnyFlatSpec with Matchers {
-  CorrelationId.init()
```
Comment (adamw) [inline backend/src/test/scala/com/softwaremill/bootzooka/test/BaseTest.scala:8]: same here, this should stay in the migrated form

File: backend/src/main/scala/com/softwaremill/bootzooka/user/UserService.scala [hunk 8]
```diff
@@ -144,9 +142,10 @@ class UserService(
       _ <- validateNewPassword()
       _ = logger.debug(s"Changing password for user: $userId")
       _ <- userModel.updatePassword(userId, User.hashPassword(newPassword))
-      confirmationEmail = emailTemplates.passwordChangeNotification(user.login)
-      _ <- emailScheduler(EmailData(user.emailLowerCased, confirmationEmail))
-    } yield ()
+    } yield {
+      val confirmationEmail = emailTemplates.passwordChangeNotification(user.login)
+      emailScheduler(EmailData(user.emailLowerCased, confirmationEmail))
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/user/UserService.scala:147]: This effect needs to be sequenced with the others - I'm surprised this worked (did tests pass)? Anyway, fixing

File: backend/src/main/scala/com/softwaremill/bootzooka/user/UserService.scala [hunk 9]
```diff
@@ -120,16 +118,16 @@ class UserService(
...
     }
 
-    def sendMail(user: User) = {
+    def sendMail(user: User): ConnectionIO[Unit] = {
       val confirmationEmail = emailTemplates.profileDetailsChangeNotification(user.login)
       emailScheduler(EmailData(user.emailLowerCased, confirmationEmail))
     }
 
-    doChange() flatMap { anyUpdate =>
+    doChange().map { anyUpdate =>
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/user/UserService.scala:126]: Here we need a .flatMap - to sequence the effects, and not ignore them. Fixing

File: backend/src/main/scala/com/softwaremill/bootzooka/util/CorrelationIdDecorator.scala [hunk 10]
```diff
@@ -49,40 +39,3 @@ object CorrelationIdDecorator {
     def extractCid(t: T): Option[String]
   }
 }
-
-/** Based on [[https://olegpy.com/better-logging-monix-1/]]. Makes the current correlation id available for logback loggers.
-  */
-class MonixMDCAdapter extends LogbackMDCAdapter {
```
[the remark this answers is not in the document]
Comment (adamw) [reply backend/src/main/scala/com/softwaremill/bootzooka/util/CorrelationIdDecorator.scala:55]: this logback adapter needs to stay :)

**Comment by adamw** [comment]
Thanks! :) I added one follow-up: #769

**Pull request merged**